src/main/java/core/basesyntax/services/impl/FruitReportServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.services.impl;
+
+import core.basesyntax.dao.Storage;
+import core.basesyntax.dto.Transaction;
+import core.basesyntax.services.FruitReportService;
+import core.basesyntax.services.TransactionParser;
+import core.basesyntax.services.Validator;
+import core.basesyntax.strategy.Operation;
+import java.util.List;
+import java.util.Map;
+
+public class FruitReportServiceImpl implements FruitReportService {
+    public static final String COMMA = ",";
+
+    @Override
+    public String createReport(List<String> listData,
+                               Map<Transaction.Operation, Operation> operationMap) {
+        Validator validator = new ValidatorImpl();
+        TransactionParser parserService = new TransactionParserImpl();
+        for (String row : listData) {
+            validator.checkInputData(listData);
+            Transaction transaction = parserService.parse(row);
+            operationMap.get(transaction.getOperation()).apply(transaction);
+        }
+        return getFormattedStringReport();
+    }
+
+    @Override
+    public String getFormattedStringReport() {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("fruit,quantity").append(System.lineSeparator());
+        for (Map.Entry<String, Integer> entry : Storage.fruitStorage.entrySet()) {
+            stringBuilder.append(entry.getKey())
+                    .append(COMMA)
+                    .append(entry.getValue())
+                    .append(System.lineSeparator());
+        }
+        return stringBuilder.toString();
+    }
```

Comment:
This is rather different logic and including it into single service and reusing one method in another one violates single responsibility principle and makes code harder to maintain and test.

Please leave getFormattedStringReport in this service under name createReport.
But method that uses strategy should be extracted to separate service and be something like
public void processInputData()
So it's work is only update Storage with input data